src/converters/comments/convertFileComments.ts

```diff
@@ -16,6 +16,10 @@ export const convertFileComments = async (
 ) => {
     const fileContent = await dependencies.fileSystem.readFile(filePath);
     if (fileContent instanceof Error) {
+        if (fileContent.code === "EISDIR") {
+            return undefined;
+        }
```

Comment:
[Structure] I actually think the approach here isn't what I'd prefer. `collectCommentFileNames` is taking in a `typescriptConfiguration` that looks like:

```json
{
  "compilerOptions": { "target": "es3" },
  "files": ["./cool/index.ts"],
  "include": ["cool"]
}
```

...but with an output `commentFileNames` like:

```js
{
  exclude: undefined,
  include: [ './cool/index.ts', 'cool' ]
}
```

(note that the original TSConfig that made that is just `{ "include": ["cool"] }`. 

I think the right approach to fix this earlier in the system would be to have **`collectCommentFileNames`** filter out directories from its output. Or maybe just not use the `"include"` at all, if `files` has the full list of files?

What do you think?

Reply:
That sounds good - I changed things so `collectCommentFileNames` handled the filtering of directories